store/search/src/main/java/org/apache/carbondata/store/worker/SearchRequestHandler.java

```diff
@@ -93,10 +93,13 @@ public ShutdownResponse handleShutdown(ShutdownRequest request) {
     List<CarbonRow> rows = new LinkedList<>();
     try {
       while (reader.nextKeyValue()) {
-        rows.add(reader.getCurrentValue());
+        // copy the data as the reader may reuse the same buffer, if unsafe is enabled
+        rows.add(new CarbonRow(reader.getCurrentValue().getData()));
```

Comment:
This commit will lead to JVM crash in my local machine. After revert, it will running success.